pydra/engine/workers.py

```diff
@@ -304,15 +300,51 @@ async def _submit_job(self, batchscript, name, checksum, cache_dir):
                     done in ["CANCELLED", "TIMEOUT", "PREEMPTED"]
                     and "--no-requeue" not in self.sbatch_args
                 ):
-                    if (cache_dir / f"{checksum}.lock").exists():
+                    if (cache_dir / f"{uid}.lock").exists():
```

Comment:
this part i don't understand. the task lock file should use the checksum, since we want the same exact jobs to be locked when run.

Reply:
I somehow missed this commend yesterday. I've tried to remove all checsksums from worker and submitter. I thought that this is what we discussed. 
But indeed checksum probably make more sense here (and likely solves other problem I have)